security2/src/main/java/zone/cogni/asquare/security2/service/PermissionService.java

```diff
@@ -0,0 +1,147 @@
+package zone.cogni.asquare.security2.service;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+import javax.annotation.PostConstruct;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class PermissionService<T extends Enum> {
+  private static final Logger log = LoggerFactory.getLogger(PermissionService.class);
+  private final Map<String, Set<String>> blacklistProperties = new HashMap<>();
+  private final Map<String, Set<String>> notBlacklistedForRole = new HashMap<>();
+  private final Map<String, String> ldap2roleName = new HashMap<>();
+  private final Map<String, Set<T>> roleName2permissions = Collections.synchronizedMap(new HashMap<>());
+  private final Class<T> enumClass;
+  private Set<T> defaultPermissions;
+  private Set<String> allBlacklistProperties;
+  @Value("${asquare.security.role_group.resource:/role-group.json}")
+  private String ldapRoleGroupResource;
+
+  public PermissionService(Class<T> enumClass) {
+    this.enumClass = enumClass;
+  }
+
+  @PostConstruct
+  @SuppressWarnings("unchecked")
+  private void init() throws Exception {
+    log.info("Using ldapRoleGroupResource '{}'", ldapRoleGroupResource);
+
+    try (InputStream inputStream = PermissionService.class.getResourceAsStream(ldapRoleGroupResource)) {
+      Map<String, String> jsonData = new ObjectMapper().enable(JsonParser.Feature.ALLOW_COMMENTS).readValue(inputStream, Map.class);
+      for (Map.Entry<String, String> entry : jsonData.entrySet()) {
+        ldap2roleName.put(StringUtils.lowerCase(entry.getKey()), StringUtils.lowerCase(entry.getValue()));
+      }
+    }
+
+    try (InputStream inputStream = PermissionService.class.getResourceAsStream("/role_access.json")) {
```

Comment:
resource location to be specified in config (do not use a default value + use config prefix used in security2 lib)

also add a bit of info log during init